### ISSUE TITLE
Calling another constructor from a constructor in a struct will decompile to "this" assignment

### ISSUE THREAD

**Issue by MulleDK19** [0xd4d/dnSpy#1287]
dnSpy v6.0.5 (.NET Framework)

The following C# source:
```csharp
public struct MyType
{
    public int Leet;

    public MyType(int dummy) : this(dummy, dummy)
    {
        this.Leet += dummy;
    }

    public MyType(int dummy1, int dummy2)
    {
        this.Leet = dummy1 + dummy2;
    }
}
```

When compiled and decompiled with dnSpy, becomes:
```csharp
public struct MyType
{
	public int Leet;

	public MyType(int dummy)
	{
		this = new MyType(dummy, dummy);
		this.Leet += dummy;
	}

	public MyType(int dummy1, int dummy2)
	{
		this.Leet = dummy1 + dummy2;
	}
}
```

While that's valid C# code, there **is** a difference.
Using `: this(...)` syntax seems to always result in copy elision, while `this = new MyType(...)` doesn't seem to ever cause copy elision.

&nbsp;

`: this(...)` syntax results in:
```csharp
.method public hidebysig specialname rtspecialname instance void .ctor(int32 dummy) cil managed
{
	.maxstack 8
	IL_0000: ldarg.0 // arg0: Load address of the MyType that's being initialized onto the stack.
	IL_0001: ldarg.1 // arg1: int32 dummy
	IL_0002: ldarg.1 // arg2:: int32 dummy
	IL_0003: call      instance void MyType::.ctor(int32, int32) // Let the other constructor initialize this instance.

	IL_0008: ldarg.0 // For stfld below.

	IL_0009: ldarg.0
	IL_000A: ldfld     int32 MyType::Leet

	IL_000F: ldarg.1
	IL_0010: add

	IL_0011: stfld     int32 MyType::Leet
	IL_0016: ret
}
```

&nbsp;

`this = new MyType(...)` results in:
```csharp
.method public hidebysig specialname rtspecialname instance void .ctor(int32 dummy) cil managed
{
	.maxstack 8
	IL_0000: ldarg.0 // For stobj below. Load address of the MyType that's being initialized onto the stack.

                         // arg0: handled by newobj below.
	IL_0001: ldarg.1 // arg1: int32 dummy
	IL_0002: ldarg.1 // arg2: int32 dummy
	IL_0003: newobj    instance void MyType::.ctor(int32, int32) // Create new MyType instance and initialize that.

	IL_0008: stobj     MyType // Copy the newly instantiated instance into the address (Address loaded at IL_0000).

	IL_000D: ldarg.0 // For stfld below.

	IL_000E: ldarg.0
	IL_000F: ldfld     int32 MyType::Leet
	
        IL_0014: ldarg.1
	IL_0015: add

	IL_0016: stfld     int32 MyType::Leet
	IL_001B: ret
}
```

&nbsp;

So while the `: this(...)` constructor call simply passes the address of the instance being initialized to the other constructor, the `this = new MyType(...)` creates a new temporary instance, and passes the address of that copy to the other constructor, then copies the memory into the original instance being initialized afterwards.

So basically, if the other constructor is invoked with `call` dnSpy should decompile it to `: this(...)`, and if it's invoked with newobj, it should decompile to `this = new MyType(...);`

**Comment by wwh1004** [comment]
i tested. it is also a bug in ilspy.
